src/ros_filter.cpp

```diff
@@ -2903,6 +2921,24 @@ namespace RobotLocalization
       }
       else
       {
+        // make pose refer to the baseLinkFrame as source
+        if (childFrame != baseLinkFrameId_)
```

Comment:
Are we sure this will work (as in, was it tested?). I feel like we should apply this transform _after_ line 2951, but the math may work out the same.

Regardless, I really only want this logic to apply to cases where `differential` or `relative` are true (as the parameter name implies), so this should go inside the `if (relative)` block below. As it's written now, this logic will apply even if `differential` and `relative` are both false.

Reply:
Hey Tom, thanks for your feedback.

What about the bug I found in my previous implementation (see [this](https://github.com/cra-ros-pkg/robot_localization/pull/728#issuecomment-1062673239) comment)? So, the same problem with the "wrong" child frame also exists without `relative` or `differential` mode. That's why, I proposed to extend the new (parameterized) behavior to all three cases (normal, `relative`, and `differential`). That makes more sense to me and it makes the implementation (as it is now) much easier. What do you think about that?

If we agree to implement the "use other child frame for pose" behavior (behind a parameter), then the test above actually does the right thing (i also tested it): whenever the parameter **is not set**, the `childFrame` is going to be `baseLinkFrameId_`, and hence the program ignores the new code; whenever the parameter **is set**, we need to correct the transformation in both cases (normal and `relative`).

Reply:
OK, then I guess we just need to make sure the parameters and variable names are sensible, as discussed above. The most critical thing is that the behavior is unchanged when the parameter is false.

Reply:
Alright, so I'll rename the parameter to `pose_use_child_frame` and double-check that the old behavior is not touched when the parameter is unset.

Reply:
Done.
I double checked all settings: normal, diff, rel in combination with an odometry topic having base_link as the pose's source and another frame as the pose's source. Each combination worked as expected. In particular, the old behavior is untouched.